lua/swampshop/vgui/preview.lua

```diff
@@ -3,24 +3,825 @@
+    self.CameraButton.Static = true
+
+    self.CameraButton.DoClick = function()
+        surface.PlaySound("UI/buttonclick.wav")
+        SS_EDITOR_ALIGNZ = not SS_EDITOR_ALIGNZ
+        self.CameraButton:SetImage(SS_EDITOR_ALIGNZ and "swampshop/view_lock.png" or "swampshop/view_tilt.png")
+    end
+
+    self.CameraButton:SetImage(SS_EDITOR_ALIGNZ and "swampshop/view_lock.png" or "swampshop/view_tilt.png")
+
+
+    self.BackButton = self:AddButton(nil, "Done", true,true)
+    self.BackButton.Static = true
+
+    self.BackButton.DoClick = function()
+        surface.PlaySound("UI/buttonclick.wav")
+        local cust = SS_CustomizerPanel
+        SS_ItemServerAction(cust.item.id, "configure", cust.item.cfg)
+        SS_CustomizerPanel:Close()  
+    end
+    --[[
+    self.SaveButton = self:AddButton("icon16/disk.png", "Save Changes", false,true)
+    self.SaveButton.Static = true
+    self.SaveButton.DoClick = function()
+        surface.PlaySound("UI/buttonclick.wav")
+        SS_ItemServerAction(SS_CustomizerPanel.item.id, "configure", SS_CustomizerPanel.item.cfg)
+    end
+    
+    self.SaveButton.Think = function(pnl)
+        local cust = SS_CustomizerPanel
+        local needsave = cust.item and cust.item._modified
+        local img = needsave and "icon16/exclamation.png" or "icon16/disk.png"
+        if(pnl:GetImage() != img)then
+            pnl:SetImage(img)
+        end
+    end
+    ]]
+
+    --[[
+    self.RevertButton = self:AddButton(nil, "Undo Changes", false,true)
+    self.RevertButton.Static = true
+    self.RevertButton.DoClick = function()
+        local cust = SS_CustomizerPanel
+        surface.PlaySound("UI/buttonclick.wav")
+        if( cust.item._modified and cust.item._cachedcfg)then
+            cust.item.cfg = table.Copy(cust.item._cachedcfg)
+            cust.item._cachedcfg = nil
+            cust.item._modified = nil
+            cust:UpdateCfg()
+            cust:SetupControls(cust.controlzone)
+        end
+    end
+    ]]
+    self.RevertButton = self:AddButton(nil, "Cancel", true,true)
+    self.RevertButton.Static = true
+    self.RevertButton.DoClick = function()
+        local cust = SS_CustomizerPanel
+        surface.PlaySound("UI/buttonclick.wav")
+        if( cust.item._modified and cust.item._cachedcfg)then
+            cust.item.cfg = table.Copy(cust.item._cachedcfg)
+            cust.item._cachedcfg = nil
+            cust.item._modified = nil
+            cust:UpdateCfg()
+ 
+        end
+        SS_CustomizerPanel:Close()  
+    end
+
+    self.ResetButton = self:AddButton(nil, "Clear Changes", false,true)
+    self.ResetButton.Static = true
+    self.ResetButton.DoClick = function()
+        local cust = SS_CustomizerPanel
+        surface.PlaySound("UI/buttonclick.wav")
+        cust.item.cfg = {}
+        cust:UpdateCfg()
+        cust:SetupControls(cust.controlzone)
+    end
+
+    
+    self.SetupSnaps = function()
```

Comment:
Can this logic either be finished or removed?

Reply:
i only disabled it because it wasn't properly updating the snap value until after recreating the gizmo